src/Components/Components/src/BindConverter.cs

```diff
@@ -496,6 +497,16 @@ private static string FormatDateTimeOffsetValueCore(DateTimeOffset value, Cultur
             return value.ToString();
         }
 
+        private static string? FormatArrayValueCore<T>(T value, CultureInfo? culture)
+        {
+            if (value is not Array array)
+            {
+                return null;
+            }
+
+            return $"[{string.Join(", ", array.Cast<object>().Select(item => $"\"{item}\""))}]";
```

Comment:
This format does not apply recursively, i.e. the elements of the array do not get formatted via `BindConverter`, and are instead converted directly to `string`s. This may need to change at some point, or additional testing may reveal that this needs to be supported before this PR is merged (e.g. binding an array of `DateTime`).

Reply:
Is casting important here and why?

Reply:
> This format does not apply recursively

It is a bit weird to be ignoring the `culture` parameter here, but it's also difficult to think of realistic cases where the *values* of options in a select would be culture-specific. Likewise it's unusual not to be respecting the bind `format` parameter in this logic, even though in most cases I don't think people would use it.

Does it add a lot of difficulty to handle those two as well?

Separately, about using the LINQ methods here - @pranavkm went though a whole campaign to remove the higher-level `System.Linq` usages from here because of concerns about them being allocatey, or other perf concerns. I think that applies here too, since this logic will allocate a lot of strings. Would it work to use a `StringBuilder` inside a `for` loop here instead of the `string.Join` and `.Cast`/`.Select`? Pranav can give more details about the perf impact if you want to know :)

Reply:
I'll look into respecting the `culture` and `format` parameters. I actually had `StringBuilder`-based implementation at first but swapped it out for this, I'm happy to switch it back!

Reply:
@SteveSandersonMS I would actually like to clarify with you what you meant regarding the `format` parameter here - the `BindFormatter<T>` delegate only takes a `CultureInfo`, so I'm not really sure where the format would come into play here (unless we made a public variant that accepted one).

It also looks like we would need to make `FormatArrayValueCore` apply the bind formatter to its elements recursively in order to use the `CultureInfo`, since `object.ToString()` does not have an overload accepting a `CultureInfo`. I think it then becomes a question of whether we're willing to accept the overhead and complexity that comes with reflection to make this work.

It's wishful thinking, but would be really neat if we could pattern match generic types and do something like
```csharp
if (T is ICollection<out TElement>)
{
  var formatter = FormatterDelegateCache.Get<TElement>();
  // ...
}
```
as an alternative to reflection, but unfortunately I think we're stuck doing reflection shenanigans if we want to make this work.

We may also be able to avoid reflection if we refactored `FormatterDelegateCache`. We could move most of `Get<T>` to a method that accepts a `Type`, since the generic-ness of `T` is not used until the enum/array/default cases. Then we could add a non-generic `BindFormatter` that wraps `BindFormatter<T>` and accepts an `object` instead of a `T`...

Let me know if you have any thoughts about this 😃

Reply:
Lots of good questions in there :)

This overlaps with https://github.com/dotnet/aspnetcore/pull/33950#discussion_r662524153 so that probably covers what I could say.